k8s/helm-chart/dynatrace-gcp-function/templates/_helpers.tpl

```diff
@@ -59,9 +59,29 @@ activeGateUrl in case of the .Values.activeGate.useExisting is
 {{- else -}}
   {{- printf "%s" .Values.dynatraceLogIngestUrl }}
 {{- end -}}
-   
-
-
 {{- end }}
 
+{{- define "activeGateHost" }}
+  {{- $agurl := (include "activeGateUrl" .)}}
+  {{- printf "%s" (split "/" (split "//" $agurl)._1)._0  }}
+{{- end }}
 
+{{/*
+noProxyUrls is used in case of the .Value.useProxy is set ALL, DT_ONLY or GCP_ONLY
+  In a case the AG will be installed the communication between pods doesn't need a proxy.
+  In a case we are using an existing AG and the proxy should be used only for GCP then the communication between Function and AG must not use the proxy.
+  In a case we are using an existing AG and the proxy should be used only for DT.
+*/}}
+{{- define "noProxyUrls"}}
+{{- if eq .Values.activeGate.useExisting "false" -}}
```

Comment:
With this condition we don't handle the case when a client choose new AG deployment and useProxy "DT_ONLY".
I agree that when new AG is created during installation, communication doesn't need a proxy but still a client can put "DT_ONLY" in values file -> then our proxy will work unpredictably/incorrect -> we won't set `googleapis.com` in noProxyUrls so we will use proxy for gcp requests.
IMO we need to adjust this condition to handle such case somehow. It's important to notice that adding `googleapis.com` to noProxyUrls (when DT_ONLY is set) is equivalent to not using proxy at all...

@karolkieglerski , @sq2gxo , @pawelsiwek what do you think? Any ideas?:)

Reply:
@dtwwolinski